system/src/main.cpp

```diff
@@ -263,19 +263,24 @@ void app_setup_and_loop(void)
     USB_USART_LineCoding_BitRate_Handler(system_lineCodingBitRateHandler);
 #endif
 
-    bool threaded = system_thread_get_state(NULL)!=0 && (system_mode()!=SAFE_MODE);
+    bool threaded = system_thread_get_state(NULL) != spark::feature::DISABLED &&
+      (system_mode()!=SAFE_MODE);
 
     Network_Setup(threaded);
 
 #if PLATFORM_THREADING
     if (threaded)
     {
-        SYSTEM_THREAD_START();
+        SystemThread.start();
         ApplicationThread.start();
     }
     else
-#endif
     {
+        SystemThread.setCurrentThread();
+        ApplicationThread.setCurrentThread();
+    }
+#endif
```

Comment:
@m-mcgowan Do you agree with this: if PLATFORM_THREADING is compiled in, but threading is disabled, the thread that runs `loop()` should return true for `isCurrentThread()` for both the application and system thread. I achieved this by marking `SystemThread` and `ApplicationThread` as attached to the current thread.